pkg/platform/kube/logProxy/elastic.go

```diff
@@ -60,10 +66,222 @@ func NewElasticLogProxy(config *platformconfig.ElasticSearchConfig) (*ElasticLog
+		}
+	}
+
+	resp, err := e.client.Search().
+		Index(e.index).
+		Request(searchRequest).
+		Perform(ctx)
+
+	if err != nil {
+		return nil, errors.Wrap(err, "Failed to execute search request")
+	}
+	return resp.Body, nil
 }
 
-func (e *ElasticLogProxy) GetFunctionReplicas(ctx context.Context, options *GetFunctionReplicaOptions) ([]string, error) {
-	return nil, nil
+func (e *ElasticLogProxy) getFunctionBaseSearchRequest(functionName string) *search.Request {
+	searchRequest := search.NewRequest()
+
+	// filters by function name custom query parameter
+	query := types.Query{
+		Bool: &types.BoolQuery{
+			Must: []types.Query{
+				{
+					Wildcard: map[string]types.WildcardQuery{
+						"kubernetes.pod.name": {
+							Value: common.Pointer(fmt.Sprintf("nuclio-%s-*", functionName)),
+						},
+					},
+				},
+				{
+					QueryString: &types.QueryStringQuery{
+						Query: e.customQueryParam,
+					},
+				},
+			},
+		},
+	}
+
+	searchRequest.Query = &query
+	return searchRequest
+}
+
+func (e *ElasticLogProxy) addTermsFilter(searchRequest *search.Request, field string, values []string) {
+	if len(values) == 0 {
+		return
+	}
+
+	fieldValues := make([]types.FieldValue, len(values))
+	for i, val := range values {
+		fieldValues[i] = types.FieldValue(val)
+	}
+
+	query := types.Query{
+		Terms: &types.TermsQuery{
+			TermsQuery: map[string]types.TermsQueryField{
+				field: types.TermsQueryField(fieldValues),
+			},
+		},
+	}
+
+	searchRequest.Query.Bool.Filter = append(searchRequest.Query.Bool.Filter, query)
+}
+
+func (e *ElasticLogProxy) addTimeFilter(request *search.Request, timeFilter *platform.TimeFilter) error {
+	if timeFilter == nil {
+		return nil
+	}
+	if err := e.addTimeSort(request, timeFilter.Sort); err != nil {
+		return errors.Wrap(err, "Failed to unmarshal sort order")
+	}
+	if timeFilter.Since != nil || timeFilter.Until != nil {
+		rangeQuery := types.DateRangeQuery{}
+		// Date fields are stored in Elasticsearch in a standard format based on ISO 8601
+		// When you use range filters (like gte or lte), the values should be in this format:
+		// "2023-11-21T14:00:00Z"
+		if timeFilter.Since != nil {
+			rangeQuery.Gte = common.Pointer(timeFilter.Since.Format(time.RFC3339Nano))
+		}
+
+		if timeFilter.Until != nil {
+			rangeQuery.Lte = common.Pointer(timeFilter.Until.Format(time.RFC3339Nano))
+		}
+
+		request.Query.Bool.Filter = append(request.Query.Bool.Filter, types.Query{
+			Range: map[string]types.RangeQuery{
+				"@timestamp": rangeQuery,
+			},
+		})
+	}
+	return nil
+}
+
+func (e *ElasticLogProxy) addTimeSort(request *search.Request, sort string) error {
+	if sort != "" {
+		var sortOrder sortorder.SortOrder
+		if err := sortOrder.UnmarshalText([]byte(sort)); err != nil {
+			return errors.Wrap(err, "Failed to unmarshal sort order")
+		}
+		request.Sort = []types.SortCombinations{
+			&types.SortOptions{
+				SortOptions: map[string]types.FieldSort{
+					"@timestamp": {
+						Order: &sortorder.Asc,
```

Comment:
Can it not be `Desc`?

Reply:
@TomerShor right, forgot to change it after testing something